### ISSUE TITLE
Sending GraphQL::Types::ISO8601DateTime over subscription gives NoMethodError(undefined method `iso8601' for "2019-06-13 14:28:40 +0000":String)

### ISSUE THREAD

**Issue by github0013** [rmosolgo/graphql-ruby#2326]
# Environments

graphql (1.9.6)
ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin16]
Rails 5.2.3
Darwin mac.local 16.7.0 Darwin Kernel Version 16.7.0: Wed Feb 27 00:29:57 PST 2019; root:xnu-3789.73.43~1/RELEASE_X86_64 x86_64

**To Reproduce**

```rb
class YourSchema < GraphQL::Schema
  use GraphQL::Subscriptions::ActionCableSubscriptions
  mutation(Types::MutationType)
  query(Types::QueryType)
  subscription(Types::SubscriptionType)
end

```

```rb
module Subscriptions
  class YourSubscriptionName < GraphQL::Schema::Subscription
    field :results, GraphQL::Types::ISO8601DateTime, null: false
    def subscribe
      # this will return the time on the initial call
      { results: Time.now }
    end
  end
end
```

```rb
# this fails
# NoMethodError(undefined method `iso8601' for "2019-06-13 14:28:40 +0000":String)
YourSchema.subscriptions.trigger("yourSubscriptionName", { }, { results: Time.now })
```

**Expected behavior**

This shouldn't have NoMethodError

```rb
YourSchema.subscriptions.trigger("yourSubscriptionName", { }, { results: Time.now })
```

**Comment by somenugget** [comment]
I had the same issue with `GraphQL::Types::ISO8601DateTime` type.

Later it appeared that `coerce_input` was not called for any `GraphQL::Schema::Scalar`'s child class.

**Comment by rmosolgo** [comment]
Thanks for the detailed report! I agree it should work like you described.

**Comment by github0013** [comment]
I think this issue relates to this issue https://github.com/rmosolgo/graphql-ruby/issues/1242 also.

**To Reproduce**

```rb
module Types
  class NonArType < Types::BaseObject
    field :id, ID, null: false
    field :name, String, null: false
  end
end

module Subscriptions
  class SubTest < GraphQL::Schema::Subscription
    field :results, Types::NonArType, null: false
    def subscribe
      # sending an OpenStruct here works fine
      { results: OpenStruct.new({id: "some_id", name: "bob" }) }
    end
  end
end

```


```rb
# if you do this, it gives an error 
BackendSchema.subscriptions.trigger(
  "subTest",
  {},
  {
    results: OpenStruct.new({ id: "some_id", name: "bob" })
  }
)
```

```
There was an exception - RuntimeError(        Failed to implement NonAr.id, tried:

        - `Types::NonArType#id`, which did not exist
        - `String#id`, which did not exist
        - Looking up hash key `:id` or `"id"` on `#<OpenStruct id="some_id", name="bob">`, but it wasn't a Hash

        To implement this field, define one of the methods above (and check for typos)
)
```

**Comment by DmitryTsepelev** [comment]
This issue is fixed in 1.9.14
